### PR TITLE
Clean up subscriptions using requestIdleCallback instead of setTimeout

### PR DESCRIPTION
**Pull request by raymondwang**
### Checklist:

- [x] If this PR contains changes to the library itself (not necessary for e.g. docs updates), please include a changeset (see [CONTRIBUTING.md](https://github.com/apollographql/apollo-client/blob/main/CONTRIBUTING.md#changesets))
- [x] If this PR is a new feature, please reference an issue where a consensus about the design was reached (not necessary for small changes)
- [x] Make sure all of the significant new logic is covered by tests

Referenced in #11214.

### CODE REVIEW

**Comment by jerelmiller** [comment]
Hey @raymondwang 👋 

Since it looks like you're wanting to [experiment](https://github.com/apollographql/apollo-client/issues/11214#issuecomment-1726289930) with whether this change improves performance, I'll go ahead and create a snapshot release you can use to install this change. Let us know if it helps!

File: src/core/ObservableQuery.ts [hunk 1]
```diff
@@ -17,6 +17,7 @@ import {
   fixObservableSubclass,
   getQueryDefinition,
 } from "../utilities/index.js";
+import { requestIdleCallback } from "../utilities/common/requestIdleCallback.js";
```
Comment (raymondwang) [inline src/core/ObservableQuery.ts:20]: I polyfilled this for compatibility with other environments (including Jest…), but exposing it via the public API for utilities seemed like overkill. Let me know if this feels like the wrong place and I'm happy to move things around.

**Comment by raymondwang** [comment]
Thanks @jerelmiller! Just tested this solution against my environment with a couple thousand subscriptions and… this change seems to make things dramatically worse. Individual unsubscribe tasks look like they're going from 0.2 - 0.5 ms to around 1.5 - 2 ms — unacceptably slow. We're converting our query hooks to `useFragment` calls, but I'll keep digging on this later this week to see if I can better optimize this.

**Comment by jerelmiller** [comment]
Dang thats a bummer! Appreciate you taking the time to try some things out! Let me know if/when you have some changes you'd like to test against a snapshot release and I or one of the team would be happy to kick off the workflow again.

**Comment by jerelmiller** [comment]
Hey @raymondwang 👋 

I'm doing a bit of a cleanup on open PRs and noticed there hasn't been much activity lately. I'm going to go ahead and close this PR. If you'd like to continue experimenting, feel free to reopen this PR and submit changes. At the very least, thanks for the exploration thus far!

**Pull request closed without merging**